src/clang-fetcher.ts

```diff
@@ -0,0 +1,135 @@
+import * as cp from 'child_process';
+import * as debug from 'debug';
+import * as fs from 'fs-extra';
+import got from 'got';
+import * as path from 'path';
+import * as tar from 'tar';
+import * as zlib from 'zlib';
+import { ELECTRON_GYP_DIR } from './constants';
+
+const d = debug('electron-rebuild');
+
+function sleep(n: number) {
+  return new Promise(r => setTimeout(r, n));
+}
+
+async function fetch(url: string, responseType: 'buffer' | 'text', retries = 3): Promise<string | Buffer> {
+  if (retries === 0) throw new Error('Failed to fetch a clang resource, run with DEBUG=electron-rebuild for more information');
+  d('downloading:', url);
+  try {
+    const response = await got.get(url, {
+      responseType,
+    });
+    if (response.statusCode !== 200) {
+      d('got bad status code:', response.statusCode);
+      await sleep(2000);
+      return fetch(url, responseType, retries - 1);
+    }
+    d('response came back OK');
+    return response.body as any;
+  } catch (err) {
+    d('request failed for some reason', err);
+    await sleep(2000);
+    return fetch(url, responseType, retries - 1);
+  }
+}
```

Comment:
Why not at least use [`got`'s `retry` option](https://github.com/sindresorhus/got#retry)? In the exception case, [`async-retry`](https://www.npmjs.com/package/async-retry) or [`p-retry`](https://www.npmjs.com/package/p-retry) would work a bit better because you'd probably want to do something like a logarithmic/exponential backoff.

Reply:
Historically I've had issues with `got`'s retry methodology erroring out instead of retrying.  This is simple enough logic